72_config_and_simplify_tls.md

```diff
@@ -0,0 +1,367 @@
+  * `max_retries`: Maximum number of retries
+  * `tpm_hash_alg`: Hash algorithm used (e.g. for PCRs): sha1, sha256, sha384, sha512
+  * `tpm_encryption_alg`:  Used encryption algorithm: ecc, rsa (Has anyone tested actually ecc?)
+  * `tpm_signing_alg`: Signing algorithms
+  * `ek_handle`: Use persisted EK handle (Do we need this option?)
+  * `run_as`: Drop privileges to a different user after start
+* `verifier.conf`: Configuration of the verifier
+  * `ip`:  IP where the verifier listens on. (Should this be a list?)
+  * `port`: Port where the verifier listens on
+  * `registrar_ip`: IP of the registrar
+  * `registrar_port`: Port of the registrar
+  * `enable_agent_mtls`:  See TLS simplification
+  * `tls_dir`: See TLS simplification
+  * `server_key`: See TLS simplification
+  * `server_cert`: See TLS simplification
+  * `trusted_ca_in`: See TLS simplification
+  * `trusted_ca_out`: See TLS simplification
+  * `database_url`: Database configuration
+  * `database_pool_sz_ovfl`: TODO: Why do we need this option?
+  * `auto_migrate_db`: Automatically update DB schema using alembic
+  * `num_workers`: Number of processes (workers) to spawn
+  * `exponential_backoff`: Use exponential backoff for retries
+  * `retry_interval`: Time to retry again or base for exponential retry if enabled
+  * `max_retries`: Maximum number of retries
+  * `quote_interval`: polling interval for getting quotes from the agent
+  * `revocations`: Section for revocation related configuration
+    * `enabled_revocation_notifications`: List of enabled methods for sending revocation notifications
+    * `zmq_ip`: Listing IP of the revocation notifier
+    * `zmq_port`: Listing port of the revocation notifier
+    * `webhook_url`: URL to send notifications to
+  * `max_upload_size`: Upload size for policies to the verifier
+  * `measured_boot_policy_name`: Name of the policy that is used for measured
+    boot
+  * `severity_labels`: List of used severity levels
+  * `severity_policy`: List of regexes that match a event ID to a severity label
+  * `ignore_tomtou_errors`: If "time of measure, time of use" (ToMToU) errors
+    should be treated as a failure. (Should this part of the IMA policy instead
+    the Keylime configuration?)
+
+### Unification of data types
+Because we the configuration is for Python we use the default Python types.
+
+* Booleans: `True` and `False`
+* Strings: Not escaped or  escaped with `"` e.g. `test = test.txt` or `test = "test with space.txt"`
+* Lists: Defined as in Python e.g. `[1, 2, "string"]`
+* Dicts: Defined as in Python e.g. `{"key": "value", "key2": 1}`
+
+We will provide a `getlist` and `getdict` that are converting those entries to
+the Python structures using `ast.literal_eval`.
+
+### Removal of the following options
+* `tls_check_hostnames`: Only in webapp code, but unused. 
+* Some database options, because URLs are the most flexible format for SQLAlchemy:
```

Comment:
I strongly support we finally remove this database-specific named options and retain DB URLS only